tests/MessagePackAnalyzer.Tests/MessagePackAnalyzerTests.cs

```diff
@@ -194,4 +196,67 @@ public class Bar : Foo
 
         await VerifyCS.VerifyCodeFixAsync(inputs, outputs);
     }
+
+    [Fact]
+    public async Task AddAttributesToMembersOfRecord()
+    {
+        string input = Preamble + /* lang=c#-test */ @"
+[MessagePackObject]
+public record Foo
+{
+    public string {|MsgPack004:Member1|} { get; set; }
+    public string {|MsgPack004:Member2|} { get; set; }
+}
+";
+
+        string output = Preamble + /* lang=c#-test */ @"
+[MessagePackObject]
+public record Foo
+{
+    [Key(0)]
+    public string Member1 { get; set; }
+    [Key(1)]
+    public string Member2 { get; set; }
+}
+";
+
+        var context = new CSharpCodeFixTest<MessagePackAnalyzer.MessagePackAnalyzer, MessagePackAnalyzer.MessagePackCodeFixProvider, DefaultVerifier>();
+        context.ReferenceAssemblies = ReferenceAssemblies.Net.Net60.WithPackages(ImmutableArray.Create(new PackageIdentity("MessagePack", "2.0.335")));
+        context.CompilerDiagnostics = CompilerDiagnostics.Errors;
+        context.SolutionTransforms.Add(static (solution, projectId) =>
+        {
+            return solution.WithProjectParseOptions(projectId, new CSharpParseOptions(languageVersion: LanguageVersion.CSharp9));
+        });
+
+        context.TestCode = input;
+        context.FixedCode = output;
+
+        await context.RunAsync();
+    }
+
+    [Fact]
+    public async Task AddAttributesToMembersOfRecordWithPrimaryCtor()
+    {
+        string input = Preamble + /* lang=c#-test */ @"
+[MessagePackObject]
+public record Foo(
+    string {|MsgPack004:Member1|},
+    string {|MsgPack004:Member2|});
+";
+
+        string output = input; // No fix for this
+
+        var context = new CSharpCodeFixTest<MessagePackAnalyzer.MessagePackAnalyzer, MessagePackAnalyzer.MessagePackCodeFixProvider, DefaultVerifier>();
+        context.ReferenceAssemblies = ReferenceAssemblies.Net.Net60.WithPackages(ImmutableArray.Create(new PackageIdentity("MessagePack", "2.0.335")));
+        context.CompilerDiagnostics = CompilerDiagnostics.Errors;
+        context.SolutionTransforms.Add(static (solution, projectId) =>
+        {
+            return solution.WithProjectParseOptions(projectId, new CSharpParseOptions(languageVersion: LanguageVersion.CSharp11));
+        });
+
+        context.TestCode = input;
+        context.FixedCode = output;
+
+        await context.RunAsync();
```

Comment:
We usually try to format this more like this:

```suggestion
        await new VerifyCS.Test
        {
            ReferenceAssemblies = ReferenceAssemblies.Net.Net60.WithPackages(ImmutableArray.Create(new PackageIdentity("MessagePack", "2.0.335"))),
            CompilerDiagnostics = CompilerDiagnostics.Errors,
            SolutionTransforms = { 
                static (solution, projectId) =>
                {
                    return solution.WithProjectParseOptions(projectId, new CSharpParseOptions(languageVersion: LanguageVersion.CSharp11));
                }),
            },
            TestCode = input, 
            FixedCode = output,
        }.RunAsync();
```

Although I see you're using a different type here than is defined as `VerifyCS` at the top of the file, so maybe this exact translation won't work. But can you see what you can do?

Reply:
Done. Indeed better readable 👍